web/react/components/mention_list.jsx

```diff
@@ -23,6 +23,11 @@ module.exports = React.createClass({
                 }
             }
         );
+        $(document).click(function() {
+            if($('#'+self.props.id).length && $('#'+self.props.id).get(0) !== $(':focus').get(0)) {
+                self.setState({mentionText: "-1"})
```

Comment:
If the document is clicked and the current item in focus is not the text box, then we don't want the mention list to show up. This is accomplished by setting the state, which is the method used throughout the react object to make the mention list disappear.